src/d3a/models/area/markets.py

```diff
@@ -15,84 +15,158 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from pendulum import DateTime # noqa
-from typing import Dict  # noqa
-
-from d3a.models.market import GridFee
-from d3a.models.market.two_sided import TwoSidedMarket
-from d3a.models.market.one_sided import OneSidedMarket
-from d3a.models.market.balancing import BalancingMarket
-from d3a.models.market import Market # noqa
-from d3a_interface.constants_limits import ConstSettings
 from collections import OrderedDict
+from typing import Dict, List, TYPE_CHECKING
+
+from d3a import constants
 from d3a.d3a_core.util import is_timeslot_in_simulation_duration
-import d3a.constants
+from d3a.models.market import GridFee, Market
+from d3a.models.market.balancing import BalancingMarket
+from d3a.models.market.one_sided import OneSidedMarket
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from pendulum import DateTime
+
+if TYPE_CHECKING:
+    from d3a.models.area import Area
 
 
 class AreaMarkets:
+    """Class that holds all markets for areas."""
+
     def __init__(self, area_log):
-        # Children trade in `markets`
         self.log = area_log
+        # Children trade in `markets`
         self.markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
         # Past markets
         self.past_markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.past_balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
-        self._indexed_future_markets = {}
+        # Settlement markets
+        self.settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.past_settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.indexed_future_markets = {}
 
     @property
-    def indexed_future_markets(self):
-        return self._indexed_future_markets
+    def all_future_spot_markets(self) -> List:
```

Comment:
Nit:
```suggestion
    def future_spot_markets(self) -> List:
```
Given the name, I would assume that it will return "all" the future spot markets in any case.

Reply:
But it does return all future markets in any case. Or did I miss something?

Reply:
Yes, what I mean is that the shorter name `future_spot_markets` means the same, and it's shorter. I think the `all_` part is not needed.

Reply:
You are right, will remove it.